Source/Scene/WebMapServiceImageryProvider.js

```diff
@@ -184,6 +184,13 @@ function WebMapServiceImageryProvider(options) {
 
   var pickFeatureResource = resource.clone();
 
+  // If the getFeatureInfo url is specified in the parameters (featureurl),
+  // we should use it instead of the url
+  if (defined(options.parameters) && defined(options.parameters.featureurl)) {
```

Comment:
@tfili Could I get your thoughts on this change? I'm not sure if `options.parameters` is the right spot for the `getFeatureInfo` url here, or if it makes sense to always replace `options.url` if it exists.

Reply:
It definitely shouldn't go in `parameters`. I would do something like the following using `options.getFeatureInfoUrl`. (`options.getFeatureInfoParameters` exists so we should follow a similar naming format)

```
  var resource = Resource.createIfNeeded(options.url);

  var pickFeatureResource = defined(options.getFeatureInfoUrl) ? Resource.createIfNeeded(options.getFeatureInfoUrl) : resource.clone();
```

Reply:
@tfili I will update the code again based on your idea. Thank you.

Reply:
Hi @ebogo1 and @tfili I have updated the code based on your suggestions and the test cases for your review. Thank you.